File: Samat_Check/FBA/portal/fba-portal/src/app/models/productActionDetail.ts
import { Product } from "../models/product";
import { ProductAction } from "../models/productAction";
import { ActionType } from "../models/actionType";
import { PaymentType } from "../models/paymentType";
import { Supplier } from "../models/supplier";
import { Tax } from "../models/tax";

export class ProductActionDetail {
	ID?: string;
	Product?: Product;
	ProductID: string;
	ProductAction?: ProductAction;
	ProductActionID: string;
	Count: number;
    Pricebuy?: number;
	Pricesell: number;
	Tax?: Tax;
    TaxID?: string;
	Reference?: string;
	Partnumber?: string;
	ActionType?: ActionType;
	ActionTypeID?: string;
	PaymentType?: PaymentType;
	PaymentTypeID?: string;
	Supplier?: Supplier;
	SupplierID?: string;
}

/*
type ProductActionDetail struct {
	ID              uuid.UUID `gorm:"primary_key;type:uuid;default:uuid_generate_v4()"`
	Product         Product
	ProductID       uuid.UUID `gorm:"type:uuid REFERENCES Product(Id)"`
	ProductAction   ProductAction
	ProductActionID uuid.UUID `gorm:"type:uuid REFERENCES ProductAction(Id)"`
	Count           int64
	Pricebuy        *float64
	Pricesell       float64
	Tax             Tax
	TaxID           uuid.UUID `gorm:"type:uuid REFERENCES Tax(Id);"`
	Reference       *string   `gorm:"type:varchar;"`
	Partnumber      *string   `gorm:"type:varchar;"`
	ActionType      ActionType
	ActionTypeID    uuid.UUID `gorm:"type:uuid REFERENCES ActionType(Id)"`
	PaymentType     PaymentType
	PaymentTypeID   uuid.UUID `gorm:"type:uuid REFERENCES PaymentType(Id)"`
	Supplier        *Supplier
	SupplierID      *uuid.UUID `gorm:"type:uuid REFERENCES Supplier(Id);"`
}
*/

File: Samat_Check/FBA/portal/fba/dbmodel/productActionDetail.go
package dbModel

import "github.com/go-openapi/strfmt"

//ProductActionDetail Детализация по операциям
// swagger:model ProductActionDetail
type ProductActionDetail struct {
	ID              strfmt.UUID4 `gorm:"primary_key;type:uuid;default:uuid_generate_v4()"`
	Product         Product
	ProductID       strfmt.UUID4 `gorm:"type:uuid REFERENCES Product(Id)"`
	ProductAction   ProductAction
	ProductActionID strfmt.UUID4 `gorm:"type:uuid REFERENCES ProductAction(Id)"`
	Count           int64
	Pricebuy        *float64
	Pricesell       float64
	Tax             Tax
	TaxID           strfmt.UUID4 `gorm:"type:uuid REFERENCES Tax(Id);"`
	Reference       *string      `gorm:"type:varchar;"`
	Partnumber      *string      `gorm:"type:varchar;"`
	ActionType      ActionType
	ActionTypeID    strfmt.UUID4 `gorm:"type:uuid REFERENCES ActionType(Id)"`
	PaymentType     PaymentType
	PaymentTypeID   strfmt.UUID4 `gorm:"type:uuid REFERENCES PaymentType(Id)"`
	Supplier        *Supplier
	SupplierID      *strfmt.UUID4 `gorm:"type:uuid REFERENCES Supplier(Id);"`
}

//TableName переопределяем имя таблицы
func (ProductActionDetail) TableName() string {
	return "productactiondetail"
}
